be/test/vec/columns/column_array_test.cpp

```diff
@@ -0,0 +1,425 @@
+        }
+    }
+}
+
+TEST_F(ColumnArrayTest, GetNumberOfDimensionsTest) {
+    // test dimension of array
+    for (int i = 0; i < array_columns.size(); i++) {
+        auto column = check_and_get_column<ColumnArray>(
+                remove_nullable(array_columns[i]->assume_mutable()).get());
+        auto check_type = remove_nullable(array_types[i]);
+        auto dimension = 0;
+        while (is_array(check_type)) {
+            auto nested_type = reinterpret_cast<const vectorized::DataTypeArray&>(*check_type)
+                                       .get_nested_type();
+            dimension++;
+            check_type = nested_type;
+        }
+        EXPECT_EQ(column->get_number_of_dimensions(), dimension)
+                << "column dimension is not equal to check_type dimension";
+    }
+}
+
+TEST_F(ColumnArrayTest, MaxArraySizeAsFieldTest) {
+    // test array max_array_size_as_field which is set to 100w
+    //  in operator[] and get()
+    for (int i = 0; i < array_columns.size(); i++) {
+        auto column = check_and_get_column<ColumnArray>(
+                remove_nullable(array_columns[i]->assume_mutable()).get());
+        auto check_type = remove_nullable(array_types[i]);
+        Field a;
+        column->get(column->size() - 1, a);
+        Array af = a.get<Array>();
+        if (af.size() > 0) {
+            Field ef = af[0];
+            for (int j = 0; j < max_array_size_as_field; ++j) {
+                af.push_back(ef);
+            }
+            std::cout << "array size: " << af.size() << std::endl;
+            auto cloned = column->clone_resized(0);
+            cloned->insert(af);
+            std::cout << "cloned size: " << cloned->size() << std::endl;
+            // get cloned offset size
+            auto cloned_offset_size =
+                    check_and_get_column<ColumnArray>(cloned.get())->get_offsets().back();
+            std::cout << "cloned offset size: " << cloned_offset_size << std::endl;
+
+            Field f;
+            // test get
+            EXPECT_ANY_THROW({ cloned->get(0, f); });
+            // test operator[]
+            EXPECT_ANY_THROW({ cloned->operator[](0); });
+        }
+    }
+}
+
+TEST_F(ColumnArrayTest, IsDefaultAtTest) {
+    // default means meet empty array row in column_array, now just only used in ColumnObject.
+    // test is_default_at
+    for (int i = 0; i < array_columns.size(); i++) {
+        auto column = check_and_get_column<ColumnArray>(
+                remove_nullable(array_columns[i]->assume_mutable()).get());
+        auto column_size = column->size();
+        for (int j = 0; j < column_size; j++) {
+            auto is_default = column->is_default_at(j);
+            if (is_default) {
+                // check field Array is empty
+                Field f;
+                column->get(j, f);
+                auto array = f.get<Array>();
+                EXPECT_EQ(array.size(), 0) << "array is not empty";
+            }
```

Comment:
You should also check size() > 0 for !is_default.

Reply:
done!